### ISSUE TITLE
Figure out SqlErrorQueryReturnedNoRows error

### ISSUE THREAD

**Issue by bendk** [mozilla/application-services#5040]
As part of the error handling, I added [code to track `QueryReturnedNoRows`](https://github.com/mozilla/application-services/blob/d163d0561fe41043491b9a19c02e66c5ab25f655/components/places/src/ffi.rs#L410) when deleting history (#4856). 

Since then I've been checking Sentry and trying to find the error, but I haven't seen it at all.  I'm not sure if it's because this issue just isn't happening anymore or because the reporting code is not working properly.

Once #5033 lands, I would expect to be getting lots of error reports.  If reporting works as expected, then it seems like #4856 is not happening anymore and we can close it.  If not, let's fix the error reporting code and get some sentry reports in.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/SYNC-3230)
┆Sprint End Date: 2022-10-28

**Comment by mhammond** [comment]
There are a few moving parts here:
* The fact that this was causing a crash was fixed in https://github.com/mozilla-mobile/android-components/pull/12112 - which would help explain why it's no longer in sentry, but it's probably still happening?
* As #4856 notes, there's one place where we explicitly return `QueryReturnedNoRows`, but rusqlite itself might also raise it. We instrumented that one explicit place, so the fact it's not appearing in the new error reporting code from Ben might just be because it is coming from somewhere else - which seems likely because it should be "impossible" for the code we instrumented to every hit that scenario.
* We're only adding new error reporting to logins in the first instance, and that error is from places.

So, the logins error reporting will verify the error reporting code is working correctly, but sadly, probably *does not* mean the error isn't being thrown, because those errors are no longer going to sentry. So I think the next action here might be to see about doing to places what we did to logins in #5033
